src/modules/Extension/Service.php

```diff
@@ -438,11 +413,14 @@ public function uninstall(\Model_Extension $ext)
 
     public function downloadAndExtract($type, $id)
     {
-        $ext = $this->di['extension'];
-        $manifest = $ext->getExtension($id, $type);
+        $latest = $this->di['extension_directory']->getLatestExtensionRelease($id);
+
+        if (!isset($latest['download_url'])) {
```

Comment:
I'd probably use `empty` instead of `isset`, since that'll check both that it's defined and will allow us to throw an exception if it's defined as an empty string for whatever reason